service/resource/certificate/update.go

```diff
@@ -2,12 +2,65 @@ package certificate
 
 import (
 	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/spf13/afero"
+	"k8s.io/client-go/pkg/api/v1"
+
+	"github.com/giantswarm/microerror"
 )
 
 func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
+	currentCertificateFiles, err := toCertificateFiles(currentState)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+
+	desiredCertificateFiles, err := toCertificateFiles(desiredState)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+
+	if currentCertificateFiles == nil || desiredCertificateFiles == nil {
+		return nil, nil, nil, nil
+	}
+
+	if !reflect.DeepEqual(currentCertificateFiles, desiredCertificateFiles) {
+		r.logger.Log("debug", "current certificates do not match desired certificates, need to update")
+		return nil, nil, &desiredCertificateFiles, nil
+	}
+
+	r.logger.Log("debug", "current certificates matches desired certificates, no update needed")
+
 	return nil, nil, nil, nil
 }
 
 func (r *Resource) ProcessUpdateState(ctx context.Context, obj, updateState interface{}) error {
+	updateCertificateFiles, err := toCertificateFiles(updateState)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for _, certificateFile := range updateCertificateFiles {
```

Comment:
I'm not taking care of removing certificates that are not used anymore yet, we can do that later. Want to get to testing it out in the wild.

Reply:
Lets make sure we track this in an issue.

Reply:
Already tracked in scraping issue.